actionpack/lib/action_controller/metal/http_authentication.rb

```diff
@@ -94,13 +94,17 @@ def authenticate(request, &login_procedure)
       end
 
       def has_basic_credentials?(request)
-        request.authorization.present? && (auth_scheme(request).downcase == 'basic')
+        request.authorization.present? && (auth_scheme(request).downcase == 'basic') && !user_name_or_password_blank?(request)
       end
 
       def user_name_and_password(request)
         decode_credentials(request).split(':', 2)
       end
 
+      def user_name_or_password_blank?(request)
```

Comment:
This name doesn't match the implementation - either it should be `user_name_and_password_blank?` or the `all?` below should be `any?`.

Also, since the only place this method is called immediately negates the value (line 97), maybe something like `user_name_or_password_present?` would be clearer.